nimare/meta/cbma/base.py

```diff
@@ -156,6 +157,7 @@ def _fit(self, dataset):
         ma_values = self._collect_ma_maps(
             coords_key="coordinates",
             maps_key="ma_maps",
+            return_type="sparse",
```

Comment:
If we are enforcing a sparse array here, will `_compute_null_approximate`, `_compute_summarystat_est`, or `_determine_histogram_bins` ever actually encounter any of the other possible datatypes? If not, then I think there's some spaghetti code we can simplify. If yes, then I'd like to try to document the circumstances where that would happen.

Reply:
Yeah, I think once we get all tests passing using the sparse array I will remove support for the other datatypes.

Reply:
Is the PR ready for this, or are you planning to do more before removing the other datatypes?

Reply:
I think it's almost ready. The [example figures](https://nimare--725.org.readthedocs.build/en/725/auto_examples/02_meta-analyses/01_plot_cbma.html#sphx-glr-auto-examples-02-meta-analyses-01-plot-cbma-py) in the documentation are similar to the ones in the [master](https://nimare.readthedocs.io/en/latest/auto_examples/02_meta-analyses/01_plot_cbma.html#sphx-glr-auto-examples-02-meta-analyses-01-plot-cbma-py) branch but are not the same. I need to look at the code to find what I'm missing, then rerun some profiling checks for ALE, KDA, and MKDADensity.

Reply:
Ah, thanks for the clarification. In that case, I'll wait until you request my review before looking through everything.

Reply:
I think I am ready to remove the support for other data types. I started doing that this morning but some tests were failing because they were created to work with numpy array. Should we remove those tests in that case?

I can also provide a table with the profiling results `ALE`, `KDA`, and `MKDADensity` against the master branch. At first glance, the use of sparse directly in the `CBMAEstimator` makes everything more efficient in terms of memory and computation time.
Also, `_run_fwe_permutation` efficiency got improved by using sparse.

Reply:
@tsalo to remove return_type="array" in favor of return_type="sparse" I would need to implement sparse in ALESubtraction too. Do you think that can be done here or in another PR?

Reply:
I think it's worth implementing it in this PR.